lib/core/src/rcMisc.cpp

```diff
@@ -1566,17 +1566,37 @@ getLocalTimeStr( struct tm *mytm, char *timeStr ) {
 }
 
 
-/* Get the current time + offset , in the  form: 2006-10-25-10.52.43 */
-/*  offset is a string of the same form  */
-/*                                               0123456789012345678 */
-void
-getOffsetTimeStr( char *timeStr, const char *offSet ) {
-    time_t myTime;
-
-    myTime = time( NULL );
-    myTime += atoi( offSet );
 
-    snprintf( timeStr, TIME_LEN, "%d", ( uint ) myTime );
+/* Get the current time + offset , and store to timeStr (understood max length TIME_LEN) in epoch seconds.
+    OffSet is a string of the form nnnn<Unit> with Unit in {'','s','m','h','d','y'}.
+    An empty Unit field implies seconds. */
+int
+getOffsetTimeStr( char *timeStr, const char *offSet )
+{
+    long seconds_multiplier = 1L;
+    static const std::map<char,long> multiplier {
+        {'\0',1}, {'s',1}, {'m',60}, {'h',3600}, {'d',24*3600}, {'y',24*3600*365}
+    };
+    time_t myTime = time( NULL );
+    const char *lTrimOffSet = offSet;
+    while(isspace(*lTrimOffSet)) {++lTrimOffSet;}               // left-trim any whitespace
+    int nonDigitLoc = strspn(lTrimOffSet,"0123456789");
+    char u = lTrimOffSet[nonDigitLoc];                          // u is value of first nondigit character
+                                                                // (Guaranteed not beyond the terminating null)
+    try {
+        seconds_multiplier = multiplier.at(u);
+    }
+    catch (std::out_of_range & e) {
+        // An unmapped, non-whitespace character as the unit field is an error.
+        if (!(isspace(u))) {
+            rodsLog(LOG_ERROR, "Incorrect delay interval <%s>; needed <nnnn[U]> with optional U in {s,m,h,d,y}", lTrimOffSet);
```

Comment:
The error message leads me to believe that I can only use a value up to and not exceeding 9,999, and cannot have whitespace between the value and the optional unit. Is that right?

Also, this function is an excellent candidate for unit tests, but I'll consider that a bonus ;D

Also, the caught exception can be const and actually doesn't need a name as it is not used in this block. Pretty minor.

Reply:
The documentation says 'nnnn' but the code actually allows more digits.  The only thing that I've done is allowed units to be used again, no new limitations, so for example 86400s = 1d and all that. I've tested beyond the year 2038 ("20y")and that works too.  I am pretty happy with it and it should pass the test suite i'm running now, and unless someone wants to educate me in the catch2 and unit tests department, probably will leave it as is...
I cannot enforce integers as @trel wanted because  test_delay_queue was too finicky; if you have any spare time you might look at  `test_delay_queue_with_long_job`. It was brittle for some reason, didn't like being an integer, so I had to leave the PLUSET field at 0.1s

Reply:
Okay, just wanted to make sure.

As for `test_delay_queue`, what you said seems fine for now. I think the tests should probably be updated but is probably outside of the scope of this issue.

Reply:
@alanking Ok, let me know what kind of issue if any we should make for updating the funky fractions (`0.1s`) in ` test_delay_queue_with_long_job` and friends.  I read through and wasn't quite clear about the intent of the tests.

Reply:
Filed #5526 for this. Thanks